src/core.js

```diff
@@ -1709,6 +1719,19 @@ define([
         this.toggleConstraintItext(mug);
     };
 
+    fn._setURLHash = function (mug) {
+        if (mug && mug.hashtagPath &&
+            mug.hashtagPath.startsWith("#") &&
+            mug.hashtagPath.indexOf('undefined') === -1 &&
```

Comment:
could this possibly cause a false positive if the name of the question actually contained "undefined"?

Reply:
Yes. Is there a canonical way of figuring out whether the question has a defined path or not? When you make a new question in a group, for example, it has a hashtagPath of `#form/group_name/undefined`.

Reply:
I think `mug.getNodeID() !== undefined` is what you want here. However, I forget what that will return for Choice questions. Also, it may not do what you want for transaction question types.